src/main.py

```diff
@@ -1,2 +1,128 @@
+from rdkit import Chem
+from fastapi import FastAPI, UploadFile
+from fastapi import status
+
+
 def substructure_search(mols, mol):
-    pass
\ No newline at end of file
+    substructure_matches = []
+    initial_mol = Chem.MolFromSmiles(mol)
+
+    for molecule in mols:
+        list_mol = Chem.MolFromSmiles(molecule, sanitize=False)
+
+        if initial_mol.HasSubstructMatch(list_mol) \
+                or list_mol.HasSubstructMatch(initial_mol):
+            substructure_matches.append(molecule)
+
+    return substructure_matches
+
+
+molecules_db = {
+    "PUBCHEM1": "NCC",
+    "PUBCHEM2": "c1ccccc1",
+    "PUBCHEM3": "CCO",
+    "PUBCHEM4": "CC(=O)O",
+    "PUBCHEM5": "CC(=O)Oc1ccccc1C(=O)O"
+}
+
+ID_COUNTER = len(molecules_db)
```

Comment:
Yes, that's true, but basically when i was testing the code, I ran into an issue.
Let's imagine we have 4 molecules: "PUBCHEM1", "PUBCHEM2", "PUBCHEM3", "PUBCHEM4".
Then I delete "PUBCHEM1", I have 3 left(2,3,4), when I create new molecule, it is going to have an id of "PUBCHEM{len+1}" which is going to be 4, but I already have "PUBCHEM4", so that created an issue and that's why I chose to work with separate ID.